src/vendorpromo/forms.py

```diff
@@ -110,7 +110,14 @@ def clean_discount_value(self):
                 raise forms.ValidationError(_("Must be a number between 1 and 99"))
         
         return discount_value
+    
+    def clean_end_date(self):
+        end_date = self.cleaned_data['end_date']
+
+        if end_date < timezone.now():
```

Comment:
end date is optional right? should this be something like 
`end_date = self.cleaned_data.get('end_date')`
`if end_date and end_date < timezone.now():`

Reply:
Thanks, you are right!!!!